### PR TITLE
Un-unfuns blood regen and adds more to it

### PR DESCRIPTION
**Pull request by Sebbe9123**
## About The Pull Request
Makes some adjustments to blood regen by making it 3 tiered based on your levels of blood.

## Why It's Good For The Game
The previous iteration made you starve when you went below 90% to regen your blood and players reported that it wasn't very fun. This aims to make the regen better when you are near full blood. It also makes blood regen better when you are close to death but with small amounts of tox.

## Changelog

:cl:
balance: Rebalances blood regen to be more player friendly above 90%
balance: Adds a tier below 50% blood to quickly bring you above 50%
/:cl:

### CODE REVIEW

**Comment by chizzy376** [comment]
> It also makes blood regen better when you are close to death but with small amounts of tox.

So if I'm correctly reading this, you're going to die faster if you are already dying? Why is this a good idea? I didn't know donating blood irl was poisoning me all this time

**Review by ivanmixo — APPROVED** [review]

File: code/modules/mob/living/blood.dm [hunk 1]
```diff
@@ -66,17 +66,34 @@
...
+					if(NUTRITION_OVERFED to INFINITY)
+						adjust_nutrition(-10)
+						blood_volume += 1 // regenerate blood quickly.
+					if(NUTRITION_HUNGRY to NUTRITION_OVERFED)
+						adjust_nutrition(-5)
+						blood_volume += 0.5 // regenerate blood slowly.
+					if(0 to NUTRITION_HUNGRY)
+						adjust_nutrition(-1)
+						blood_volume += 0.1 // Regenerate blood VERY slowly.
+			if(BLOOD_VOLUME_SURVIVE to BLOOD_VOLUME_OKAY)
```
Comment (QualityVan) [inline code/modules/mob/living/blood.dm:83]: The toxloss from this is very high, and low blood already has damage applications innately. Beyond that, I'm not convinced that dividing nutrition effects between low and very low blood is helpful either for comprehension or gameplay. The safe to normal buffer zone and base regen is good and should address the main issue.

File: code/modules/mob/living/blood.dm [hunk 2]
```diff
@@ -66,17 +66,34 @@
...
-				if(NUTRITION_HUNGRY to NUTRITION_OVERFED)
-					adjust_nutrition(-5)
-					blood_volume += 0.5 // regenerate blood slowly.
-				if(0 to NUTRITION_HUNGRY)
-					adjust_nutrition(-1)
-					blood_volume += 0.1 // Regenerate blood VERY slowly.
+		switch(blood_volume)
+			if(BLOOD_VOLUME_SAFE to BLOOD_VOLUME_NORMAL) //Passively regens blood very slowly from 90% to 100% without a tradeoff.
+				blood_volume += 0.1
+			if(BLOOD_VOLUME_OKAY to BLOOD_VOLUME_SAFE) //Regens blood from 50% ish to 90% using nutrition.
```
Comment (QualityVan) [inline code/modules/mob/living/blood.dm:72]: OKAY is 60% of normal.

**Review by QualityVan — CHANGES_REQUESTED** [review]

**Review by QualityVan — APPROVED** [review]

**Comment by ivanmixo** [comment]
Seems okay

**Pull request merged**